### ISSUE TITLE
BarTouchTooltipData cannot be displayed under Y-axis 0, Y-axis <0

### ISSUE THREAD

**Issue by Tate-zwt** [imaNNeo/fl_chart#1462]
**Describe the bug**
When the Y-axis is less than 0, the BarTouchTooltipData prompt can only be displayed on the 0-point baseline. In fact, I want it to be above the -X point. This should be a BUG. Please take a look. Thank you very much!
**To Reproduce**
maxY = 180 
minY = -600
```
BarTouchTooltipData(
                direction: TooltipDirection.top,
                tooltipHorizontalAlignment: FLHorizontalAlignment.center,
                fitInsideHorizontally: true, 
                fitInsideVertically: true, 
                tooltipBgColor: Colors.transparent,
                tooltipPadding: EdgeInsets.zero,
                tooltipMargin: 0,
                getTooltipItem: ((group, groupIndex, rod, rodIndex) {
                  return (rod.fromY != 0 && rod.toY != 0)
                      ? BarTooltipItem(
                          rod.toY.toInt().toString(),
                          10.textStyle(MYColors.colorB1B1B1, bold: true),
                        )
                      : BarTooltipItem('', 10.textStyle(Colors.transparent, bold: true));
                }),
              )
```
**Screenshots**
If applicable, add screenshots, or videoshots to help explain your problem.
<img width="311" alt="image" src="https://github.com/imaNNeo/fl_chart/assets/10485427/d2778901-509c-4d41-8444-8384e0cab356">

**Versions**
 - which version of the Flutter are you using?
 Flutter 3.10.5
 - which version of the FlChart are you using?
 fl_chart: ^0.63.0

I hope BarTouchTooltip supports displaying at the bottom of the bar chart，Looking forward to your reply！

**Comment by imaNNeo** [comment]
I didn't understand it. Can you also provide me a reproducible code? Thanks!

**Comment by Tate-zwt** [comment]
<img width="407" alt="image" src="https://github.com/imaNNeo/fl_chart/assets/10485427/d738aaea-c201-4e51-8ecc-34bd778fbc37">

Hello, the distance on the picture, according to the distance set by the code, should be equivalent. However, when running, the result is not equal. At present, it is because if it is lower than -Y, it can only be displayed at the 0 point position. I I think the prompt should be based on the top of the histogram, thank you!

The following code example can be run：
```
import 'package:fl_chart/fl_chart.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter fl_chart Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter fl_chart Demo'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[_setupChart()],
        ),
      ),
    );
  }

  Widget _setupChart() {
    Widget? chart;
    chart = BarChart(
      barChartData(),
      swapAnimationDuration: Duration.zero,
    );

    return AspectRatio(
      aspectRatio: 339 / 147,
      child: Row(
        children: [
          Expanded(
            child: Stack(
              children: [
                chart,
              ],
            ),
          ),
        ],
      ),
    );
  }

  BarChartData barChartData() {
    return BarChartData(
      barGroups: [
        _createGroupDataSleep(
          0,
          -250,
          -50,
        ),
        _createGroupDataSleep(
          1,
          -280,
          -100,
        ),
        _createGroupDataSleep(
          2,
          -210,
          10,
        )
      ],
      maxY: 100,
      minY: -300,
      barTouchData: BarTouchData(
          enabled: true,
          touchTooltipData: BarTouchTooltipData(
            direction: TooltipDirection.top,
            tooltipHorizontalAlignment: FLHorizontalAlignment.center,
            fitInsideHorizontally: true,
            fitInsideVertically: true,
            tooltipBgColor: Colors.transparent,
            tooltipPadding: EdgeInsets.zero,
            tooltipMargin: 0,
            getTooltipItem: ((group, groupIndex, rod, rodIndex) {
              return (rod.fromY != 0 && rod.toY != 0)
                  ? BarTooltipItem(rod.toY.toString(), const TextStyle(color: Colors.black, fontSize: 10))
                  : BarTooltipItem('', const TextStyle(color: Colors.black, fontSize: 10));
            }),
          )),
      alignment: BarChartAlignment.spaceAround,
      titlesData: const FlTitlesData(
        show: true,
        leftTitles: AxisTitles(sideTitles: SideTitles(showTitles: false)),
        topTitles: AxisTitles(sideTitles: SideTitles(showTitles: false)),
        rightTitles: AxisTitles(sideTitles: SideTitles(showTitles: false)),
        bottomTitles: AxisTitles(sideTitles: SideTitles(showTitles: false)),
      ),
      gridData: const FlGridData(
        show: false,
      ),
    );
  }

  BarChartGroupData _createGroupDataSleep(
    int x,
    double fromY,
    double toY, {
    double width = 15,
  }) {
    return BarChartGroupData(
      x: x,
      groupVertically: true,
      barRods: [
        BarChartRodData(
          fromY: fromY,
          toY: toY,
          color: Colors.amber,
          width: width,
          borderRadius: BorderRadius.circular(0),
        ),
      ],
      showingTooltipIndicators: [0],
    );
  }
}
```

**Comment by Tate-zwt** [comment]
I hope BarTouchTooltip supports displaying at the bottom of the bar chart!

**Comment by imaNNeo** [comment]
Fixed in [0.66.0](https://github.com/imaNNeo/fl_chart/releases/tag/0.66.0)
